pyanaconda/dbus/interface.py

```diff
@@ -0,0 +1,323 @@
+
+            # Check the kind of the parameter
+            if parameters[name].kind != Parameter.POSITIONAL_OR_KEYWORD:
+                raise DBusSpecificationError("Only positional or keyword arguments are allowed.")
+
+            # Check if the type is defined.
+            if name not in type_hints:
+                raise DBusSpecificationError("Parameter %s doesn't have defined type." % name)
+
+            yield name, type_hints[name], self.DIRECTION_IN
+
+        # Is the return type defined?
+        if self.RETURN_PARAMETER not in type_hints:
+            return
+
+        # Is the return type other than None?
+        if type_hints[self.RETURN_PARAMETER] is None:
+            return
+
+        yield self.RETURN_PARAMETER, type_hints[self.RETURN_PARAMETER], self.DIRECTION_OUT
+
+    def _is_method(self, member):
+        """Is the class member a DBus method?"""
+        return inspect.ismethod(member) or inspect.isfunction(member)
+
+    def _generate_method(self, member, member_name):
+        """Generate method defined by given class member.
+
+        :param member: a method object
+        :param member_name: a name of the method
+        :return: a method element
+        """
+        method = self.xml_generator.get_method_element(member_name)
+
+        # Process the parameters.
+        for name, type_hint, direction in self._iterate_parameters(member):
+            # Create the parameter element.
+            parameter = self.xml_generator.create_parameter(name, get_dbus_type(type_hint), direction)
+            # Add the element to the method element.
+            self.xml_generator.add_child(method, parameter)
+
+        return method
+
+    def _is_property(self, member):
+        """Is the class member a DBus property?"""
+        return isinstance(member, property)
+
+    def _generate_property(self, member, member_name):
+        """Generate DBus property defined by class member.
+
+        :param member: a property object
+        :param member_name: a property name
+        :return: a property element
+        """
+        access = None
+        type_hint = None
+
+        try:
+            # Process the setter.
+            if member.fset:
+                [(_, type_hint, _)] = self._iterate_parameters(member.fset)
+                access = self.ACCESS_WRITE
+
+            # Process the getter.
+            if member.fget:
+                [(_, type_hint, _)] = self._iterate_parameters(member.fget)
+                access = self.ACCESS_READ
+
+        except ValueError:
+            raise DBusSpecificationError("Property %s has invalid parameters." % member_name)
+
+        # Property has both.
+        if member.fget and member.fset:
+            access = self.ACCESS_READWRITE
+
+        if access is None:
+            raise DBusSpecificationError("Property %s is not accessible." % member_name)
+
+        return self.xml_generator.create_property(member_name, get_dbus_type(type_hint), access)
```

Comment:
Ahh sorry, I've looked on the `create_parameter` instead of `create_property` my bad.